### ISSUE TITLE
MD2 animation doesn't look smooth when it's loaded from serialized file

### ISSUE THREAD

**Issue by froggy96** [Rajawali/Rajawali#1426]
First of all, I really appreciate all you folks here for the greatness of the framework!

Here goes my issue:
I tested md2 parser/renderer according to the rajawail example (tag: v0.9)
First, I parsed ogro.md2 to get it serialized and saved into a file:

    MD2Parser parser = new MD2Parser(resources, textureManager, resourceId);
    parser.parse();
    mObject = (VertexAnimationObject3D) parser.getParsedAnimationObject();
    MeshExporter exporter = new MeshExporter(mObject);
    exporter.export(exportFilename, ExportType.SERIALIZED);

After that, I read the serialized file into a VertexAnimationObject3D object with texture:

    ObjectInputStream ois;
    ois = new ObjectInputStream(mContext.getResources().openRawResource(R.raw.ogro));
    mObject = new VertexAnimationObject3D((SerializedObject3D)ois.readObject());
    mObject.setMaterial(new DiffuseMaterial());
    mTexture = BitmapFactory.decodeResource(mContext.getResources(), R.drawable.ogro);
    mObject.addTexture(mTextureManager.addTexture(mTexture, TextureType.DIFFUSE));
    ois.close();

then I use this by using `addChild()`.
The object shows well, the animation goes too, but the animation looks like `not animating every frame`, which means you see all the keyframes with it but that's all - no other inter-frame animations between keyframes.
This results frame-dropping look and feel. I tried with various settings of mObject.setFps as well as renderer.setFrameRate in vain.
When I load ogro.md2 with MD2parser, it just looks perfect.

Is there any missing action from me here?

**Comment by froggy96** [comment]
Have anyone reproduced the situation with the code above? Is it just a feature of the framework?
Thanks,

**Comment by jwoolston** [comment]
I have not used MD2 animations before, and don't have the time to look into it at the moment, especially since the serialized objects have been removed for the `v1.0` release and you are using the `v0.9` branch. I can't think of any reason why the framework would do this but I know numerous issues with MD2 have been fixed in the master branch. You might try posting on the G+ community, there are a number of users of the `v0.9` tag there.

**Comment by froggy96** [comment]
I have found the answer!!!

I should have created DiffuseMaterial like this:

    mObject.setMaterial(new DiffuseMaterial(true));

From the MD2parser and DiffuseMaterial source, I found the constructor of DiffuseMaterial with no parameter does call it with 'false' which means the material is not for 'animation'.

Applying the above, the animation looks perfect just as I load an MD2 file.

Closing this ticket. Thanks jwoolston for your kindness too.
